### PR TITLE
Feat: refactor + update dependencies

### PR DESCRIPTION
**Pull request by malik672**
added new errors  and updated to the latest version for sdk-core

## Summary by CodeRabbit

- **New Features**
	- Upgraded the `uniswap-sdk-core` dependency to a release candidate version, improving functionality and stability.
	- Introduced a new dependency, `rustc-hash`, enhancing hashing capabilities.

- **Bug Fixes**
	- Improved error handling in various methods for better clarity and robustness.

## Summary by CodeRabbit

- **New Features**
	- Upgraded the `uniswap-sdk-core` dependency to a release candidate version, improving functionality and stability.
	- Introduced a new dependency, `rustc-hash`, enhancing hashing capabilities.

- **Bug Fixes**
	- Improved error handling in various methods for better clarity and robustness.

- **Documentation**
	- Minor formatting adjustments made to `README.md` for improved readability.

- **Tests**
	- Added new tests to validate functionality and reliability of the `Trade` struct and its methods.

### CODE REVIEW

File: src/entities/trade.rs [hunk 1]
```diff
@@ -500,6 +504,39 @@ mod tests {
...
+
+        #[test]
+        fn fails_with_insufficient_reserves() {
+            let result = Trade::new(
+                Route::new(vec![EMPTY_PAIR_0_1.clone()], TOKEN0.clone(), TOKEN1.clone()),
+                CurrencyAmount::from_raw_amount(TOKEN0.clone(), 100).unwrap(),
+                TradeType::ExactInput,
+            );
+            assert!(result.is_err());
+        }
```
Comment (shuhuiluo) [inline src/entities/trade.rs]: I don't know where these came from. Follow the tests in v2-sdk.
https://github.com/Uniswap/v2-sdk/blob/main/src/entities/trade.test.ts

File: src/entities/trade.rs [hunk 2]
```diff
@@ -548,4 +585,162 @@ mod tests {
...
+                &mut result,
+            )
+            .unwrap();
+
+            assert!(!result.is_empty());
+            assert_eq!(result[0].route.pairs.len(), 1);
+            assert_eq!(result[0].route.path[0], TOKEN0.clone());
+            assert_eq!(result[0].route.path[1], TOKEN2.clone());
+        }
+    }
```
Comment (shuhuiluo) [inline src/entities/trade.rs]: Where's the rest of the tests from Uniswap? I don't need tests made up by Copilot. You're not saving my time if I have to check every unit test is migrated.
https://github.com/Uniswap/v2-sdk/blob/main/src/entities/trade.test.ts#L114-L191

**Review by shuhuiluo — CHANGES_REQUESTED** [review]
Why don't you make a PR for updating dependencies, a PR for error handling, a PR for adding tests in `trade`?

**Comment by malik672** [comment]
> Why don't you make a PR for updating dependencies, a PR for error handling, a PR for adding tests in `trade`?

i think a pr for error handling and updating dependencies should be together

**Comment by shuhuiluo** [comment]
Continue after #6. If you want to extend V2's `Error` enum, add a member `CoreError` to wrap SDK Core's `Error`.

**Pull request closed without merging**